sdk/tracing/azure-core-tracing-opencensus/src/test/java/com/azure/core/tracing/opencensus/OpenCensusTracerTest.java

```diff
@@ -51,27 +183,92 @@ public void startSpanOverloadNullPointerException() {
         openCensusTracer.start("", Context.NONE, null);
     }
 
+    //add tests for number of child spans created and how parent span relation
+
+    @Test
+    public void addLinkTest() {
+        // Arrange
+        // Create a child-parent link between multiple spans
+        RecordEventsSpanImpl existingSpan = (RecordEventsSpanImpl) tracer.spanBuilder("existing-span").startSpan();
+        Context traceContext = tracingContext.addData(SPAN_CONTEXT, existingSpan.getContext());
+        RecordEventsSpanImpl parentSpanImpl = (RecordEventsSpanImpl) parentSpan;
+
+        // Act
+        openCensusTracer.addLink(traceContext);
+
+        //Assert
+        // TODO: existing -> parentSpanImpl comes after after add Link, confirm this behavior?
+        Assert.assertEquals(parentSpanImpl.getPrev(), existingSpan);
```

Comment:
@lmolkova I do not remember this very clearly but I think we introduced something called as a `Link.fromSpanContext` to mimic otel's behavior of adding Link.

But the way it should have worked is adding a link after the parentSpan, correct?

Reply:
With opencensus, links are added one the span after span is created, with opentelemetry - before creation on builder.

What do you mean by 'after the parentSpan'?

Reply:
So the way we have the link is, we have parentSpan followed by existing span.

Reply:
Removed this and now checking if the two spans are related by the same TraceId